### PR TITLE
update readmes

### LINKED ISSUE #71: Enhance Documentation
Provide a little more detail/background in the project README such as...
  - how it differs from the current eventing-contrib kafka implementation
  - current state of the project (some history and organization as we move towards eventing-contrib compliance / structure)
  - components and helm and other oddities ; )

### PR DESCRIPTION
**Pull request by travis-minke-sap**
Quick pass at enhancing the documentation - current / more detail / etc...

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,80 +1,173 @@
...
+supports a single instance of the dispatcher/consumer.  Further, no ordering 
+guarantees on the consumption side are provided which is required in certain 
+use cases.  The reference implementation was also based on the 
+[Sarama Go Client](https://github.com/Shopify/sarama) which has limitations 
+compared to the [Confluent Go Client](https://github.com/confluentinc/confluent-kafka-go), 
+and the [librdkafka C library](https://github.com/edenhill/librdkafka) on which 
+it is based.  Finally, the ability to expose Kafka configuration was very 
+limited, and we needed the ability to customize certain aspects of the Kafka 
+Topics / Producers / Consumers.
+
```
Comment (phamilton) [inline README.md:28]: Should probably add sentence or two about supporting a variety of kafka providers

File: README.md [hunk 2]
```diff
@@ -1,80 +1,173 @@
...
+### Messaging Guarantees
 
-An event sent to a `KafkaChannel` is guaranteed to be persisted and processed if a 202 response is received by the sender.
-Events are partitioned in Kafka by the Cloud Event attribute `subject` when present or randomly when absent.  Events
-in each partition are processed in order, with an at least once guarantee. If a full cycle of retries for a given
-subscription fails, the event is ignored and processing continues with the next event.
+An event sent to a `KafkaChannel` is guaranteed to be persisted and processed 
+if a 202 response is received by the sender.  Events are partitioned in Kafka
+by the Cloud Event attribute `subject` when present or randomly when absent.  
+Events in each partition are processed in order, with an at least once guarantee.
```
Comment (phamilton) [inline README.md]: The partitioning logic is actually a little more complicated.. first it checks the field defined in the CloudEvent partitioning extension (https://github.com/cloudevents/spec/blob/master/extensions/partitioning.mdhttps://github.com/cloudevents/spec/blob/master/extensions/partitioning.md) called 'partitionkey' if that does not exist tries 'subject' and finally falls back to random.

File: README.md (hunk 1 above)
Comment (travis-minke-sap) [reply README.md:28]: 👍

File: README.md (hunk 2 above)
Comment (travis-minke-sap) [reply README.md]: Yeah, that was Davy's original text and I hadn't considered it too closely.  I'll add the detailed breakdown - thanks!

**Review by phamilton — APPROVED** [review]
LGTM

**Pull request merged**